src/main/java/de/tum/in/www1/artemis/domain/Course.java

```diff
@@ -452,4 +456,10 @@ public String toString() {
                 + ", color='" + getColor() + "'" + ", courseIcon='" + getCourseIcon() + "'" + ", registrationEnabled='" + isRegistrationEnabled() + "'" + "'"
                 + ", presentationScore='" + getPresentationScore() + "}";
     }
+
+    @JsonIgnore
+    public Set<Exercise> getInterestingExercisesForAssessmentDashboards() {
+        return getExercises().stream().filter(exercise -> exercise instanceof TextExercise || exercise instanceof ModelingExercise || exercise instanceof FileUploadExercise
```

Comment:
I don't think that `AsessmentType` is used consistently until now, but for the future this might be a good option. However, we would then need to check all database entries. I think the way I do it here can also be problematic, because we still have entries with `NULL` in the database that actually have the assessment type automatic. So I should probably check for `MANUAL` or `SEMI_AUTOMATIC`